File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {

    m := make(map[int]int)

    for i1, val := range(nums){
        needed := target - val
        
        if i2, exists := m[needed]; exists {
            return []int{i1, i2}
        }

        m[val] = i1
    }

    return []int{}
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        checked_nums = {}
        for i, num in enumerate(nums):
            needed = target - num
            if needed in checked_nums:
                return [checked_nums[needed], i]
            checked_nums[num] = i